addons/mrp/models/mrp_workcenter.py

```diff
@@ -388,16 +388,17 @@ def action_archive(self):
             }
         return res
 
-    def _get_capacity(self, product):
-        product_capacity = self.capacity_ids.filtered(lambda capacity: capacity.product_id == product)
-        return product_capacity.capacity if product_capacity else self.default_capacity
-
-    def _get_expected_duration(self, product_id):
-        """Compute the expected duration when using this work-center
-        Always use the startup / clean-up time from specific capacity if defined.
-        """
-        capacity = self.capacity_ids.filtered(lambda p: p.product_id == product_id)
-        return capacity.time_start + capacity.time_stop if capacity else self.time_start + self.time_stop
+    def _get_capacity(self, product, unit, default_capacity=1):
+        capacity = self.capacity_ids.filtered(lambda capacity: capacity.product_id == product and capacity.product_uom_id == product.uom_id)
+        if not capacity:
+            capacity = self.capacity_ids.filtered(lambda capacity: not capacity.product_id and capacity.product_uom_id == unit)
+            if not capacity:
+                capacity = self.capacity_ids.filtered(lambda capacity: not capacity.product_id and capacity.product_uom_id == product.uom_id)
```

Comment:
```python
capacity = self.capacity_ids.sorted(lambda c: (
    c.product_id == product and c.product_uom_id == product.uom_id,
    not c.product_id and c.product_uom_id == unit,
    not c.product_id and c.product_uom_id == product.uom_id,
)[:1]
```
to avoid triple loop

Reply:
done, but seems less clear with the `not` required to have True at first (and avoid `reversed`)